wallet/chainntfns.go

```diff
@@ -174,6 +181,44 @@ func (w *Wallet) ChainSwitch(ctx context.Context, forest *SidechainForest, chain
 
 			// Add the block hash to the notification.
 			chainTipChanges.AttachedBlocks = append(chainTipChanges.AttachedBlocks, n.Hash)
+
+			if birthdayState != nil && (birthdayState.SetFromHeight || birthdayState.SetFromTime) {
+				if birthdayState.SetFromTime && n.Header.Timestamp.After(birthdayTime) {
+					bh := n.Header.PrevBlock
+					birthdayState.Hash = bh
+					birthdayState.Height = n.Header.Height - 1
+					birthdayState.SetFromTime = false
+					if err := w.txStore.SetBirthState(dbtx, birthdayState); err != nil {
+						return err
+					}
+					if err := w.txStore.UpdateProcessedTxsBlockMarker(dbtx, &bh); err != nil {
+						return err
+					}
+					log.Infof("Set wallet birthday to block %d (%v).",
+						n.Header.Height-1, bh)
```

Comment:
You can unify all cases by doing the test once (`if bs != nil && ((bs.setFromHeight && ...) || (bs.setFromTime && ...))`), then using the same code to set all fields (because it's the same block either way).

Reply:
I combined them.